packages/pages/src/dev/server/ssr/generateTestData.ts

```diff
@@ -127,17 +127,27 @@ export const generateTestDataForPage = async (
     });
 
     childProcess.on("close", () => {
+      let parsedData: any;
       if (testData) {
-        testData = JSON.parse(testData.trim());
+        try {
+          parsedData = JSON.parse(testData.trim());
+        } catch (e) {
+          stdout.write(
+            `\nUnable to parse test data from command: \`${command}${args.join(
+              " "
+            )}\``
+          );
+          resolve(null);
```

Comment:
Is this what's actually fixing the crash?

Reply:
the general try/catch is preventing it, but resolve(null) lets us actually indicate that we couldn't provide the data so that the caller can handle that case more elegantly.